### PR TITLE
add close_conn command to close a given client connection

### PR DESCRIPTION
**Pull request by tharanga**
helps to evict zombie or malicious clients in real time.

### CODE REVIEW

**Comment by tharanga** [comment]
We can put this behind a config parameter if this is not widely applicable (or not desired in some situations).

File: memcached.c [hunk 1]
```diff
@@ -3633,6 +3633,31 @@ static void process_stat(conn *c, token_t *tokens, const size_t ntokens) {
...
+    if (conn == NULL) {
+        out_string(c, "ERROR Connection not found");
+    } else if (conn->state == conn_listening) {
+        out_string(c, "ERROR Listening connection");
+    } else if (IS_UDP(conn->transport) && conn->state == conn_read) {
+        out_string(c, "ERROR UDP connection");
+    } else if (conn->sfd == c->sfd) {
+        out_string(c, "ERROR Current connection");
+    } else {
+        conn_close(conn);
```
Comment (tharanga) [inline memcached.c]: Should we close the connections asynchronously?

Comment (dormando) [reply memcached.c]: haven't looked closely yet; if you do it the same way the idle connection kicker works it should be safe. IIRC that was a signal back to the owner thread to close it out?

Comment (tharanga) [reply memcached.c]: Yes it is. Let me do it that way.

**Comment by dormando** [comment]
Kind of raises a question I should've asked already; what's the use case for this that the idle thread doesn't do for you? Are you finding conns stuck in other states?

**Comment by tharanga** [comment]
This is expected to help on evicting zombie or malicious/buggy clients in real time. Idle timeout helps zombies when the specified timeout is reached.

**Comment by dormando** [comment]
what're the connections doing though? Stuck in a loop fetching an item, stuck halfway through an upload, idle, etc?

**Comment by tharanga** [comment]
Some connections are idle, clients keep on re-connecting and leave connections behind. These will eventually get cleaned up by the idle timeout, but until then they pile up. Some clients are abusive, they are active and we want to abort them.
I looked at the state machine and realized we should implement a way to interrupt the state machine and let it close connections. Closing a connection while a worker is serving could be ugly, specially on TLS.
For the moment we'll use iptables to block such clients. I will close this PR and re-open it if there's a clean way to close connections.

**Pull request closed without merging**